### PR TITLE
[v1.4] [NCL-3955] add support to lookup artifact based on checksums

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [NCL-3955] Implemented artifact endpoint with support to lookup based on checksums
* [NCL-3955] Added integration tests for new artifact enpoint

### CODE REVIEW

**Comment by thescouser89** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

File: rest/src/main/java/org/jboss/pnc/rest/provider/ArtifactProvider.java [hunk 1]
```diff
@@ -169,6 +173,13 @@ private String getPublicUrl(Artifact artifact) {
         return queryForCollection(pageIndex, pageSize, sortingRsql, query, withDependantBuildRecordId(buildRecordId));
     }
 
+    public CollectionInfo<ArtifactRest> getAll(int pageIndex, int pageSize, String sortingRsql, String query,
+            Optional<String> sha256, Optional<String> md5, Optional<String> sha1) {
+        logger.debug("HELLO + sha1: " + sha1 + " sha256:" + sha256 + " md5: " + md5);
```
Comment (jbartece) [inline rest/src/main/java/org/jboss/pnc/rest/provider/ArtifactProvider.java]: Use better debug message

Comment (michalovjan) [reply rest/src/main/java/org/jboss/pnc/rest/provider/ArtifactProvider.java]: forgot to cherry-pick last commit, my bad

**Comment by michalovjan** [comment]
retest this please

**Pull request merged**